crates/sui-protocol-config/src/snapshots/sui_protocol_config__test__version_1.snap

```diff
@@ -5,6 +5,7 @@ expression: "ProtocolConfig::get_for_version(cur)"
 version: 1
 feature_flags:
   package_upgrades: false
+  gas_buckets_sub_and_storage_computation: false
```

Comment:
this makes sense I guess but I was surprised by it. Basically as we update the struct the version1 comes back with different fields, so this would have to be updated on every new version?
@mystenmark

Reply:
ah, yes, i didn't think of this. It does make sense but its unfortunate that we can't have a blanket "old version snaps never change" rule.

There might be a serde attr we can use that omits false values in yaml. 

If that doesn't work, we might be able to use `Option<()>` instead of bool for flags, because the snapshot serializer will not (i believe) write out entries for fields with `None`. Its a bit strange but since the weirdness will be hidden behind setters maybe its ok.

Reply:
Confirmed. The following works fine:

```
#[derive(Default, Clone, Serialize)]
struct FeatureFlags {
    // Add feature flags here, e.g.:
    // new_protocol_feature: bool,
    package_upgrades: bool,

    // New flags should have these serde attrs so that they do not break snapshot tests.
    #[serde(skip_serializing_if = "is_false")]
    #[serde(default)]
    new_flag: bool,
}

fn is_false(b: &bool) -> bool {
    !*b
}
```

Reply:
do you want me to add that now?
keep in mind that this version is not going to last and it's only temporary here.
Let me know

Reply:
lets test it, this is good practice for mainnet.